Database/QuestieDB.lua

```diff
@@ -1,31 +1,6 @@
 ---@class QuestieDB
----@field npcCompilerTypes table
----@field npcCompilerOrder table
----@field npcKeys table
----@field npcKeysReversed table
----@field questCompilerTypes table
----@field questCompilerOrder table
----@field questKeys table
----@field questKeysReversed table
----@field objectCompilerTypes table
----@field objectCompilerOrder table
----@field objectKeys table
----@field objectKeysReversed table
----@field itemCompilerTypes table
----@field itemCompilerOrder table
----@field itemKeys table
----@field itemKeysReversed table
----@field npcData table
----@field questData table
----@field objectData table
----@field itemData table
----@field sortKeys table
----@field private private table
----@field private _itemAdapterQueryOrder table @temporary, until we remove the old db funcitons
----@field private _objectAdapterQueryOrder table @temporary, until we remove the old db funcitons
----@field private _questAdapterQueryOrder table @temporary, until we remove the old db funcitons
----@field private _npcAdapterQueryOrder table @temporary, until we remove the old db funcitons
 local QuestieDB = QuestieLoader:CreateModule("QuestieDB")
+QuestieDB.private = QuestieDB.private or {}
```

Comment:
Why would that be necessary?

Reply:
The weirdness i do with the .private is for the ide to be able to understand that it is defined elsewhere and make it actually link the objects together. Because it knows a private field exists but it doesn't otherwise understand what is in it.
It doesn't actually have any impact on the actual running of the code it's purely for the IDE